cmake/Modules/FindAmgXSolver.cmake

```diff
@@ -0,0 +1,74 @@
+# - Try to find AmgX
+# Once done this will define
+#
+#  AMGXSOLVER_FOUND        - system has PETSc
+#  AMGXSOLVER_INCLUDES     - the PETSc include directories
+#  AMGXSOLVER_LIBRARIES    - Link these to use PETSc
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+#
+
+#cmake_policy(VERSION 3.3)
+
+if( NOT PETSC_FOUND )
+  return()
+endif()
+
+set(AMGX_VALID_COMPONENTS
+    C
+    CXX)
+
+SET(CUDA_DIR "$ENV{CUDA_DIR}" CACHE PATH "The path to CUDA.")
+find_package(CUDA)
+
+if(${CUDA_FOUND})
+  SET(CUDA_LIBRARIES ${CUDA_LIBRARIES} ${CUDA_CUBLAS_LIBRARIES} ${CUDA_cusparse_LIBRARY} ${CUDA_cusolver_LIBRARY})
+  #MESSAGE("-- Finding CUDA - Success")
```

Comment:
When used AMGX in needed additional libraries libdl, librt? Maybe this changed and I should recheck...